src/scatra_ele/4C_scatra_ele_calc_sti_diffcond.cpp

```diff
@@ -716,11 +722,16 @@ void Discret::Elements::ScaTraEleCalcSTIDiffCond<distype>::mat_fourier(
 )
 {
   // extract material parameters from Soret material
-  const std::shared_ptr<const Mat::FourierIso> matfourier =
-      std::static_pointer_cast<const Mat::FourierIso>(material);
+  const std::shared_ptr<const Mat::Fourier> matfourier =
+      std::static_pointer_cast<const Mat::Fourier>(material);
+
   densn = densnp = densam = matfourier->capacity();
-  diff_manager()->set_isotropic_diff(matfourier->conductivity(), 0);
-}  // Discret::Elements::ScaTraEleCalcSTIDiffCond<distype>::mat_soret
+
+  std::vector<double> k = matfourier->conductivity();
```

Comment:
```suggestion
 const std::vector<double>& k = matfourier->conductivity();
```

Not essential but there is no reason to make a copy. On the other hand, the hacky code you modified here (not yours) has more serious problems.